From cdd3d90b0caea855754e455cb1b3e880f6435670 Mon Sep 17 00:00:00 2001
Message-Id: <cdd3d90b0caea855754e455cb1b3e880f6435670.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sat, 27 Sep 2014 14:58:20 +0530
Subject: [PATCH 312/312] REVERTME [VPG]: drm/i915: Skip wait for Gfx, if Gfx
 clock is already On

If GFX clk is already ON during suspend. Skip waiting for GFX to go down
and then forcing GFX clk on again. Instead return success to continue
with runtime suspend operation.

We skip the GFX  check for cherryview only.

This patch is REVERTME, With latest BIOS, we see GFX CLK On,
when we enter runtime suspend. Revert the patch once fixed.

Issue: GMINL-1917
Change-Id: I8168c2c719cb43e727357b945cfbfbdc65631f58
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 598e449..6bd2c58 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -1219,11 +1219,17 @@ int vlv_force_gfx_clock(struct drm_i915_private *dev_priv, bool force_on)
 	int err;
 
 	val = I915_READ(VLV_GTLC_SURVIVABILITY_REG);
-	WARN_ON(!!(val & VLV_GFX_CLK_FORCE_ON_BIT) == force_on);
+	/* WARN_ON only for the Valleyview */
+	WARN_ON((!!(val & VLV_GFX_CLK_FORCE_ON_BIT) == force_on) &&
+			!IS_CHERRYVIEW(dev_priv->dev));
 
 #define COND (I915_READ(VLV_GTLC_SURVIVABILITY_REG) & VLV_GFX_CLK_STATUS_BIT)
 	/* Wait for a previous force-off to settle */
 	if (force_on) {
+		/* FIXME: On CHV: GFX clock is ON by default. */
+		if (IS_CHERRYVIEW(dev_priv->dev))
+			return 0;
+
 		err = wait_for(!COND, 20);
 		if (err) {
 			DRM_ERROR("timeout waiting for GFX clock force-off (%08x)\n",
-- 
1.7.9.5

